tools/molecule.spec

```
@@ -0,0 +1,59 @@
+%global srcname molecule
+%global pypiname molecule
+%global pkgname python3-%{srcname}
+%global forgeurl https://github.com/ansible-community/%{srcname}
+
+Name:    %{pkgname}
+Version: 0.0.0
+Release: 1%{?dist}
+Summary: ...
+%forgemeta
+URL:       %{forgeurl}
+Source:    %{forgesource}
```

It's common for downstreams to use PyPI-published sdists as the source of truth, they differ from Git archives, especially since untagged Git versions will not have enough versioning metadata + they may have differing contents.

Instead, it's best to use the `%{pypi_source}` macro:
```suggestion
Source0:    %{pypi_source}
```

- https://docs.fedoraproject.org/en-US/packaging-guidelines/SourceURL/#_python_packages_pypi
- https://docs.fedoraproject.org/en-US/packaging-guidelines/Python/#_source_files_from_pypi


@hroncok am I missing anything?

---

You are missing the packit bit. Packit patches the source in, so the value might not even matter.

As for whether using sdist tarball or github archive is better -- neither one is better, they are different. Both have advantages and disadvantages.

---

But in case we enable Packit to send PRs to distgits of downstreams, would it matter?

---

I guess packit PRs replace the downstream spec file with the upstream one, so it would. But the name of the spec file here differs from the Fedora spec file name, so it might not even work. I have very little experience with keeping spec files upstream (I consider it bad practice).

---

Well, the whole point of Packit is to allow upstream testing and keep it as close as possible to downstream. I don't see a problem to rename this file to match what Fedora uses.